airbyte-webapp/src/components/experiments/SpeedyConnection/hooks/use-experiment-speedy-connection-experiment.ts

```diff
@@ -0,0 +1,18 @@
+import { useExperiment } from "hooks/services/Experiment";
+import { useGetCloudWorkspace } from "packages/cloud/services/workspaces/CloudWorkspacesService";
+import { useCurrentWorkspace } from "services/workspaces/WorkspacesService";
+
+export const useExperimentSpeedyConnection = () => {
+  const workspace = useCurrentWorkspace();
+  const cloudWorkspace = useGetCloudWorkspace(workspace.workspaceId);
+  const isVariantEnabled = useExperiment("onbarding.speedyConnection", false);
+
+  const isTrial = Boolean(cloudWorkspace.trialExpiryTimestamp);
+  const timestamp = localStorage.getItem("exp-speedy-connection-timestamp");
+
+  const expiredOfferDate = timestamp ? JSON.parse(timestamp) : 0;
```

Comment:
```suggestion
  const expiredOfferDate = timestamp ? Number(timestamp) : 0;
```

We should try to avoid JSON.parse and JSON.strnigify as much as possible. Those methods are really slow, and reduce type-safteness since the result could in theory be anything. If we know we want to parse a string to a number here, using the `Number` constructor is the better option.